src/main.rs

```diff
@@ -212,10 +212,6 @@ async fn start_processes(args: Input) {
         );
     }
 
-    // TODO: Is this the best way of doing this?
```

Comment:
So without going into _too_ much detail on it, you often need a loop of some description for keeping an async application running.

Typically you would put this somewhere right down the bottom where you kick off your handling logic. A very basic example:

```rust
#[tokio::main]
async fn main() {
    let socket = TcpListener::bind("127.0.0.1:8888").await?;
    loop {
        match socket.accept().await {
            Err(socket_error) => eprintln!("Socket error: {}", socket_error),
            Ok((mut stream, _)) => {
                tokio::spawn(async move {
                    stream.do_stuff_with_it().await;
                });
            }
        }
    }
}
```

In this case, you've set your loop up at the end. Could try moving this into the `main()`, or perhaps adjusting how you start things off for your workers.